tests/snuba/search/test_backend.py

```diff
@@ -575,10 +577,20 @@ def test_last_seen_filter_with_environment(self):
 
         results = self.backend.query(
             self.project,
+            date_to=self.group1.last_seen + timedelta(days=1),
             environment=self.environments['development'],
             last_seen_from=self.group1.last_seen,
             last_seen_from_inclusive=False,
         )
+        assert set(results) == set()
+
+        results = self.backend.query(
+            self.project,
+            date_to=self.group1.last_seen + timedelta(days=1),
+            environment=self.environments['development'],
+            last_seen_from=self.group1.last_seen,
+            last_seen_from_inclusive=True,
+        )
```

Comment:
The above found a subtle bug in the old hash style. I think the subquery/JOIN for issues was ignoring timestamps, so it was querying for data past the `end`. Using the `issue` API caught the bug and so I have to query accordingly.